src/transformStateWithClones.js

```diff
@@ -64,8 +64,44 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, transforms) {
-  // write code here
+  // const stateClone = {};
+
+  // for (const key in state) {
+  //   stateClone[key] = state[key];
+  // }
+
+  const stateClone = { ...state };
+
+  const arr = [];
+
+  for (const key of transforms) {
+    switch (key.operation) {
+      case 'addProperties':
+        Object.assign(stateClone, key.properties);
+        break;
+
+      case 'removeProperties':
+        for (const prop of key.properties) {
+          delete stateClone[prop];
+        }
+        break;
+
+      case 'clear':
+        for (const prop in stateClone) {
+          delete stateClone[prop];
+        }
+        break;
+
+      default:
+        // do nothing
```

Comment:
Don't leave 'default' state empty. Add at least 'break'.
It is useful to add 'default' case because there may be a lot of different kinds of problems so there always should be a default case to break the function's execution and exclude unpredictable consequences.